### ISSUE TITLE
Issue with ServerEventsClient - stopped flag

### ISSUE THREAD

**Issue by Nness** [ServiceStack/Issues#316]
[Restart method](https://github.com/ServiceStack/ServiceStack/blob/master/src/ServiceStack.Client/ServerEventsClient.cs#L338)

Restart method only check stopped before `sleep`, then it didn't check again after `sleep`. Where in some case, client might already force stopped. 

One of approach will be same as stop, to introduce `InternalStart`, where `InternalStart()` will verify stopped flag, while `Start()` will change stopped flag to true.

Also will be good if we can make stopped flag visible but not editable.

Eg:
```
private bool stopped = true;
public bool Stopped
{
    get { return stopped; }
}
```

**Comment by mythz** [comment]
Added in this commit: https://github.com/ServiceStack/ServiceStack/commit/3b58f2ad77155f0ff2112e9edba28a583b67b4b0

Change is now available in v4.0.43+ [on MyGet](https://github.com/ServiceStack/ServiceStack/wiki/MyGet).

**Comment by Nness** [comment]
Cool thx.
